### PR TITLE
Feat/table settings storage

### PR DESCRIPTION
**Pull request by wcbzero**
* Adds a new api version with authentication required
* Adds authentication to GUI
* Migrates api versions v1 and v2 to fastapi
* Adds table settings persistence to local storage
* Adds user menu
* Adds settings modal (from user menu), that allows the user to set the desired timezone and date time format.
  * image
  * Screenshot 2024-11-25 at 10 20 10 AM
* Adds icons to clickable fields in task instances table
  * Screenshot 2024-11-25 at 10 20 36 AM
* Changes http requests to use useQuery

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,6 +42,12 @@ docsource/api
 
 /jobmon_gui/node_modules/
 /corpora/
+<<<<<<< HEAD
```
Comment (blsmxiu47) [inline .gitignore:45]: conflict markers

File: jobmon_gui/src/components/navigation/PageNavigation.tsx [hunk 2]
```diff
@@ -104,6 +108,8 @@ export default function PageNavigation({children}: PropsWithChildren) {
     const [open, setOpen] = useState(false);
     const navigate = useNavigate();
     const [drawerOpen, setDrawerOpen] = useState(false);
+    const {user} = useContext(AuthContext)
```
Comment (blsmxiu47) [inline jobmon_gui/src/components/navigation/PageNavigation.tsx:111]: Just asking out of curiosity - is there a reason to `useContext` over zustand for auth here?

Comment (wcbzero) [reply jobmon_gui/src/components/navigation/PageNavigation.tsx:111]: This is just how it was initially implemented in other apps, and I haven't found a compelling reason to change it.

File: .gitignore (hunk 1 above)
Comment (wcbzero) [reply .gitignore]: Fixed.

**Review by everdyke — APPROVED** [review]

**Pull request merged**